java-2016/homework-9/IOhomework/src/Testing.java

```diff
@@ -0,0 +1,146 @@
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("CZ",expected.get(2).getCountryCode());
+    }
+
+    @Test
+    public void readSkiTimeFromValidFile1(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes.csv");
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("30:10",expected.get(2).getMinuteSeconds());
+    }
+
+    @Test
+    public void readSkiTimeFromValidFile2(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes.csv");
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("30:27",expected.get(0).getMinuteSeconds());
+    }
+
+    @Test
+    public void readShootingRangeFromValidFile1(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes.csv");
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("xxxox",expected.get(0).getFirstShootingRange());
+    }
+
+    @Test
+    public void readShootingRangeFromValidFile2(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes.csv");
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("xxxxx",expected.get(2).getSecondShootingRange());
+    }
+
+    @Test
+    public void readShootingRangeFromValidFile3(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes.csv");
+        ReadFile read = new ReadFile();
+        read.readFile(file);
+        List<Athlete> expected = read.getAthletes();
+
+        Assert.assertEquals("xxxxo",expected.get(1).getThirdShootingRange());
+    }
+
+    @Test()
+    public void readInvalidFile(){
+        File file = new File("C:" + File.separator + "Users" + File.separator + "Ovi" + File.separator + "Desktop" + File.separator + "Curs" + File.separator + "IOFiles" + File.separator + "Athletes!!!.csv");
+        Assert.assertFalse(file.exists());
+    }
+
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
you cannot really differentiate between the two because of the the bad exception message you have. One exception should be file not found when file.exists() is checked and the other one file.length() just signals an empty file